### PR TITLE
[Refactor]: SwitchingAccountRouter Tests

### PR DESCRIPTION
**Pull request by marcomaddy1982**
## What's new in this PR?

SwitchingAccountRouter coverage tests written using the changes made in the [PR](https://github.com/wireapp/wire-ios-sync-engine/pull/1332)

### Dependencies

sync-engine: https://github.com/wireapp/wire-ios-sync-engine/pull/1332

### CODE REVIEW

File: Wire-iOS Tests/SwitchingAccountRouterTests.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    override func tearDown() {
+        sut = nil
+        super.tearDown()
+    }
+    
+    func testThatSwitchAccountAlertIfPresented() {
+        // WHEN
+        sut.confirmSwitchingAccount { isConfirmed in }
```
Comment (David-Henner) [inline Wire-iOS Tests/SwitchingAccountRouterTests.swift:39]:
```suggestion
        sut.confirmSwitchingAccount { _ in }
```

File: Wire-iOS Tests/SwitchingAccountRouterTests.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        sut = MockSwitchingAccountRouter()
+
+    }
+
+    override func tearDown() {
+        sut = nil
+        super.tearDown()
+    }
+    
+    func testThatSwitchAccountAlertIfPresented() {
```
Comment (typfel) [inline Wire-iOS Tests/SwitchingAccountRouterTests.swift:37]:
```suggestion
    func testThatAlertIsPresented_WhenConfirmSwitchingAccountIsInvoked() {
```

File: Wire-iOS/Sources/UserInterface/SelfProfile/AccountSelectorController.swift [hunk 3]
```diff
@@ -66,22 +66,19 @@ final class AccountSelectorController: UIViewController {
...
-            AppDelegate.shared.appRootRouter?.confirmSwitchingAccount { (confirmed) in
-                if confirmed {
-                    ZClientViewController.shared?.conversationListViewController.dismiss(animated: true, completion: {
-                        AppDelegate.shared.mediaPlaybackManager?.stop()
-                        SessionManager.shared?.select(account)
-                    })
-                }
-            }
-        }
+        guard
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/SelfProfile/AccountSelectorController.swift:69]: These changes doesn't related to the main refactoring

File: Wire-iOS/Sources/SwitchingAccountRouter.swift [hunk 4]
```diff
@@ -18,36 +18,25 @@
...
+        presentSwitchAccountAlert(completion: completion)
     }
-    
-    // MARK: - Private Implementation
-    private func presentSwitchAccountAlert(with completion: @escaping (Bool) -> Void) {
+}
+
+// MARK: - SwitchingAccountAlertPresenter
+extension SwitchingAccountRouter {
+    @objc
```
Comment (typfel) [inline Wire-iOS/Sources/SwitchingAccountRouter.swift:38]: @objc is unnecessary?

File: Wire-iOS/Sources/SwitchingAccountRouter.swift [hunk 5]
```diff
@@ -18,36 +18,25 @@
...
+typealias SwitchingAccountRouterProtocol = SessionManagerSwitchingDelegate & SwitchingAccountAlertPresenter
+
+protocol SwitchingAccountAlertPresenter {
+    func presentSwitchAccountAlert(completion: @escaping (Bool) -> Void)
 }
 
+class SwitchingAccountRouter: SwitchingAccountRouterProtocol { }
+
 // MARK: - SessionManagerSwitchingDelegate
-extension SwitchingAccountRouter: SessionManagerSwitchingDelegate {
```
Comment (typfel) [inline Wire-iOS/Sources/SwitchingAccountRouter.swift:33]: A nice thing about this is that Xcode will suggest that to add any missing delegate methods directly into this extension.

**Review by typfel — CHANGES_REQUESTED** [review]

File: Wire-iOS/Sources/SwitchingAccountRouter.swift (hunk 4 above)
Comment (marcomaddy1982) [reply Wire-iOS/Sources/SwitchingAccountRouter.swift:38]: yes it is unfortunately.

File: Wire-iOS/Sources/UserInterface/SelfProfile/AccountSelectorController.swift (hunk 3 above)
Comment (marcomaddy1982) [reply Wire-iOS/Sources/UserInterface/SelfProfile/AccountSelectorController.swift:69]: they are somehow, cause I moved the checks about the ongoing call in the sync engine so they are not necessary anymore there there and moreover I wanted to remove the super ugly exposed method of the `AppRootRouter`

```swift
AppDelegate.shared.appRootRouter?.confirmSwitchingAccount 
```

P.s. tested many time and seems to work fine.

File: Wire-iOS Tests/SwitchingAccountRouterTests.swift [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import XCTest
+@testable import Wire
+
+final class SwitchingAccountRouterTests: XCTestCase {
+    
+    var sut: MockSwitchingAccountRouter!
```
Comment (johnxnguyen) [inline Wire-iOS Tests/SwitchingAccountRouterTests.swift:24]: The combination of terminologies "system under test" and "mock" is an oxymoron, I suggest to rename `MockSwitchingAccountRouter` to something like `TestableSwitichingAccountRouter`. We want to express that it is indeed a `SwitchingAccountRouter` and not some fake version of it.

Comment (marcomaddy1982) [reply Wire-iOS Tests/SwitchingAccountRouterTests.swift]: Done

**Review by typfel — APPROVED** [review]

**Pull request merged**